cmd/clusterctl/client/get_kubeconfig_test.go

```diff
@@ -17,40 +17,97 @@ limitations under the License.
 )
 
 func Test_clusterctlClient_GetKubeconfig(t *testing.T) {
+	clusterName := "workload-cluster"
 	configClient := newFakeConfig()
-	kubeconfig := cluster.Kubeconfig{Path: "kubeconfig", Context: "mgmt-context"}
-	clusterClient := newFakeCluster(cluster.Kubeconfig{Path: "cluster1"}, configClient)
-
-	// create a clusterctl client where the proxy returns an empty namespace
-	clusterClient.fakeProxy = test.NewFakeProxy().WithNamespace("").WithFakeCAPISetup()
-	badClient := newFakeClient(configClient).WithCluster(clusterClient)
+	objects := test.FakeCAPISetupObjects()
+	kubeconfig := cluster.Kubeconfig{Path: "kubeconfig"}
 
 	tests := []struct {
 		name      string
 		client    *fakeClient
 		options   GetKubeconfigOptions
 		expectErr bool
+		want      string
 	}{
 		{
-			name:      "returns error if unable to get client for mgmt cluster",
+			name:      "returns error if unable to get client for workload cluster",
 			client:    fakeEmptyCluster(),
 			expectErr: true,
 		},
 		{
-			name:      "returns error if unable namespace is empty",
-			client:    badClient,
-			options:   GetKubeconfigOptions{Kubeconfig: Kubeconfig(kubeconfig)},
+			name: "returns error if namespace is empty",
+			client: func() *fakeClient {
+				clusterClient := newFakeCluster(kubeconfig, configClient)
+
+				// create a clusterctl client where the proxy returns an empty namespace
+				clusterClient.fakeProxy = test.NewFakeProxy().WithNamespace("").WithFakeCAPISetup()
+				return newFakeClient(configClient).WithCluster(clusterClient)
+			}(),
+			options:   GetKubeconfigOptions{Kubeconfig: Kubeconfig(kubeconfig), UserKubeconfig: true},
 			expectErr: true,
 		},
+		{
+			name: "returns the default user kubeconfig",
+			client: func() *fakeClient {
+				clusterClient := newFakeCluster(kubeconfig, configClient).
+					WithObjs(append(objects, getKubeconfigSecret(true, clusterName))...)
+				return newFakeClient(configClient).WithCluster(clusterClient)
+			}(),
+			options: GetKubeconfigOptions{
+				Kubeconfig:          Kubeconfig(kubeconfig),
+				Namespace:           "default",
+				WorkloadClusterName: clusterName,
+				UserKubeconfig:      true,
+			},
+			expectErr: false,
+			want:      getKubeconfig(clusterName, true),
+		},
+		{
+			name: "returns the system kubeconfig when the secret for user kubeconfig is not available",
+			client: func() *fakeClient {
+				clusterClient := newFakeCluster(kubeconfig, configClient).
+					WithObjs(append(objects, getKubeconfigSecret(false, clusterName))...)
+				return newFakeClient(configClient).WithCluster(clusterClient)
+			}(),
+			options: GetKubeconfigOptions{
+				Kubeconfig:          Kubeconfig(kubeconfig),
+				Namespace:           "default",
+				WorkloadClusterName: clusterName,
+				UserKubeconfig:      true,
+			},
+			expectErr: false,
+			want:      getKubeconfig(clusterName, false),
+		},
+		{
+			name: "returns the system kubeconfig when GetKubeconfigOptions.UserKubeconfig is set to false",
```

Comment:
Can you please add both user kubeconfig and system kubeconfig to the list of fake object. When the `UserKubeconfig` is set to true we want the function to return the system kubeconfig even if the user kubeconfig is present.